generators/csharp/sdk/src/wrapped-request/WrappedRequestGenerator.ts

```diff
@@ -46,6 +46,9 @@ export class WrappedRequestGenerator extends FileGenerator<CSharpFile, SdkCustom
             type: csharp.Class.ClassType.Record
         });
 
+        const service = this.context.getHttpServiceOrThrow(this.serviceId);
+        // If the service transport is grpc and the endpoint transport is not http, we need to generate a proto request
+        const isProtoRequest = service.transport?.type === "grpc" && this.endpoint.transport?.type !== "http";
```

Comment:
Thoughts on building out a helper for this on the `context`? That way we can drop the comment and reuse it elsewhere later if need be. Plus, we can someday leverage the encoding field in this function more naturally.

```typescript
const isProtoRequest = this.context.endpointHasProtoRequest({ service, endpoint: this.endpoint });
```